### ISSUE TITLE
How to edit intermediate output "a_dist"

### ISSUE THREAD

**Issue by centiLinda** [StanfordASL/Trajectron-plus-plus#81]
Hi, thanks for your great work and the effort to maintain the codebase!

I refer to `p_y_xz()` which produces the model output:
https://github.com/StanfordASL/Trajectron-plus-plus/blob/1031c7bd1a444273af378c1ec1dcca907ba59830/trajectron/model/mgcvae.py#L868-L886

If I understand correctly, `a_sample` and `a_dist` are velocity sample and distribution. If we choose to integrate dynamics as follows in the config file:
`"dynamic": {"PEDESTRIAN": {"name": "SingleIntegrator", "distribution": true, "limits": {}}}, "pred_state": {"PEDESTRIAN": {"position": ["x", "y"]}}`
The output `y_dist` will be the position because it uses `integrate_distribution()`. If we choose NOT to integrate dynamics, the output would be the velocity instead.

In both cases, the `sampled_future` will always be positions, because it all uses `integrate_samples()`. 

My question is, why is `a_dist` the velocity distribution? Is it defined somewhere? If so, can I change it to be, for example, acceleration distribution instead?

**Comment by BorisIvanovic** [comment]
Hi @centiLinda , thank you for the kind words, and for your question!

Effectively, `a_dist` is whatever your dynamics require it to be. In the case above, `a_dist` is the velocity distribution purely because we chose single integrator dynamics to convert it to position. However, if you wished to have `a_dist` be the acceleration distribution, then you would want to use double integrator dynamics to convert it to positions. As another example, if you wanted to make `a_dist` to be steering and longitudinal acceleration, you would use unicycle dynamics.

Overall, the model has no idea from the outset what `a_dist` represents, it's just outputting some values which are then passed into one more processing function before a loss is computed. Here, the "processing function" is dynamics and the loss is the comparison to the ground truth positions. Then, via repeated backpropagation through the dynamics, the model learns to make `a_dist` the action distribution corresponding to the chosen dynamics (this is actually why we named it `a_dist` -- it is the action distribution for the chosen dynamics model!).

I hope this helps!

**Comment by zjysteven** [comment]
Hi @BorisIvanovic, this is a long shot but I have a follow-up question in this thread. Like mentioned above, `y_dist` will be the  action distribution when `self.hyperparams['dynamic'][self.node_type]['distribution']` is false. In this case, I don't see how consequentially the log likelihood and loss are computed correctly. More specifically, the `labels` here are positions but `y_dist` is action distribution, and this mismatch confuses me. Not sure if I'm missing anything.
https://github.com/StanfordASL/Trajectron-plus-plus/blob/1031c7bd1a444273af378c1ec1dcca907ba59830/trajectron/model/mgcvae.py#L942-L944

-----------
**Edit:** Never mind. I figured out what I'm missing: When `y_dist` is the action distribution (e.g. velocity) the corresponding `pred_state` is also velocity, so there is no mismatch.

**Comment by zjysteven** [comment]
In most configs (e.g., `int_ee`, `int_ee_me` for NuScenes) `hyperparams['dynamic'][self.node_type]['distribution']` is true, and the action distribution will be integrated to get position distribution, which makes perfect sense to me. However in `vel_ee` (constant velocity baseline) `hyperparams['dynamic'][self.node_type]['distribution']` is false, and single integrator is used as the dynamic model, which means that `a_dist` and `y_dist` will be the velocity distribution. This is where I'm confused.
